src/components/ExpandableText/index.tsx

```diff
@@ -0,0 +1,57 @@
+import { Typography } from "antd";
+import { ArrowDownOutlined, ArrowUpOutlined } from "@ant-design/icons";
+import "./style.css";
+import { useEffect, useRef, useState } from "react";
+import {
+    DEFAULT_DESCRIPTION_HEIGHT,
+    TEXT_BOTTOM_MARGIN,
+} from "../../constants";
+
+const { Paragraph } = Typography;
+
+interface ExpandableTextProps {
+    text: string;
+    setCurrentHeight: (height: number) => void;
+}
+
+const expandSymbol = (
+    <span>
+        expand <ArrowDownOutlined />
+    </span>
+);
+
+const collapseSymbol = (
+    <span>
+        collapse <ArrowUpOutlined />
+    </span>
+);
+
+const ExpandableText = ({ text, setCurrentHeight }: ExpandableTextProps) => {
+    const [isExpanded, setIsExpanded] = useState<boolean>(false);
+    const ref = useRef<HTMLParagraphElement>(null);
+
+    useEffect(() => {
+        setCurrentHeight(
+            (ref.current?.clientHeight || DEFAULT_DESCRIPTION_HEIGHT) +
+                TEXT_BOTTOM_MARGIN
+        );
+    }, [isExpanded, setCurrentHeight]);
```

Comment:
these are the main changes, I needed to have a way of tracking after the re-render was complete to then get the new height of the element